src/chain/chain_state.cpp

```diff
@@ -295,28 +248,18 @@ size_t chain_state::retarget_height(size_t height, uint32_t forks,
         retargeting_interval : retarget_distance(height, retargeting_interval));
 }
 
-size_t chain_state::bip9_bit0_height(size_t height, uint32_t forks)
+size_t chain_state::bip9_bit0_height(size_t height,
+    const config::checkpoint& bip9_bit0_active_checkpoint)
 {
-    const auto testnet = !script::is_enabled(forks, rule_fork::difficult);
-    const auto regtest = !script::is_enabled(forks, rule_fork::retarget);
-
-    const auto activation_height =
-        testnet ? testnet_bip9_bit0_active_checkpoint.height() :
-        regtest ? regtest_bip9_bit0_active_checkpoint.height() :
-            mainnet_bip9_bit0_active_checkpoint.height();
-
+    const auto activation_height = bip9_bit0_active_checkpoint.height();
     // Require bip9_bit0 hash at heights above historical bip9_bit0 activation.
     return height > activation_height ? activation_height : map::unrequested;
 }
 
-size_t chain_state::bip9_bit1_height(size_t height, uint32_t forks)
+size_t chain_state::bip9_bit1_height(size_t height,
+    const config::checkpoint& bip9_bit1_active_checkpoint)
 {
-    const auto testnet = !script::is_enabled(forks, rule_fork::difficult);
-
-    const auto activation_height = testnet ?
```

Comment:
Looks like this patch is also removing a silent bug because the case for `regtest` is missing as it was done in line 303.